lib/document/stateTransition/validation/structure/validateDocumentsBatchTransitionStructureFactory.js

```diff
@@ -0,0 +1,203 @@
+          dataContractId,
+          ownerId,
+          rawTransition.$type,
+          rawTransition.$entropy,
+        );
+
+        if (rawTransition.$id !== documentId) {
+          result.addError(
+            new InvalidDocumentTransitionIdError(rawTransition),
+          );
+        }
+
+        // validate entropy
+        if (!entropy.validate(rawTransition.$entropy)) {
+          result.addError(
+            new InvalidDocumentTransitionEntropyError(rawTransition),
+          );
+        }
+      });
+
+    const enrichedBaseDataContract = enrichDataContractWithBaseSchema(
+      dataContract,
+      baseTransitionSchema,
+    );
+
+    const enrichedDataContractsByActions = {
+      [AbstractDocumentTransition.ACTIONS.CREATE]: enrichDataContractWithBaseSchema(
+        enrichedBaseDataContract,
+        createTransitionSchema,
+      ),
+      [AbstractDocumentTransition.ACTIONS.REPLACE]: enrichDataContractWithBaseSchema(
+        enrichedBaseDataContract,
+        replaceTransitionSchema,
+      ),
+    };
+
+    // Validate schema of CREATE and REPLACE transitions
+    createTransitions
+      .concat(replaceTransitions)
+      .forEach((rawTransition) => {
+        // validate document schema
+        const documentSchemaRef = dataContract.getDocumentSchemaRef(rawTransition.$type);
+
+        const additionalSchemas = {
+          [dataContract.getJsonSchemaId()]: enrichedDataContractsByActions[rawTransition.$action],
+        };
+
+        result.merge(
+          validator.validate(
+            documentSchemaRef,
+            rawTransition,
+            additionalSchemas,
+          ),
+        );
+      });
+
+    // validate DELETE transitions
+    deleteTransitions
+      .forEach((rawTransition) => {
+        result.merge(
+          validator.validate(
+            baseTransitionSchema,
+            rawTransition,
+          ),
+        );
+      });
+
+    // Find duplicate documents by type and ID
+    const duplicateTransitions = findDuplicatesById(rawStateTransition.transitions);
+    if (duplicateTransitions.length > 0) {
+      result.addError(
+        new DuplicateDocumentTransitionsError(duplicateTransitions),
+      );
+    }
+
+    // Find duplicate transitions by unique indices
+    const duplicateTransitionsByIndices = findDuplicatesByIndices(
+      rawStateTransition.transitions,
+      dataContract,
+    );
+
+    if (duplicateTransitionsByIndices.length > 0) {
+      result.addError(
+        new DuplicateDocumentTransitionsError(duplicateTransitionsByIndices),
+      );
+    }
+
+    if (!result.isValid()) {
+      return result;
+    }
+
+    // User must exist and confirmed
+    result.merge(
+      await validateIdentityExistence(
+        ownerId,
+      ),
+    );
+
+    if (!result.isValid()) {
+      return result;
+    }
+
+    const stateTransition = new DocumentsBatchTransition(rawStateTransition);
+
+    // Verify ST signature
+    stateTransition
```

Comment:
I think we don't need it anymore since we set these data in the constructor above.